Clarify that multiline LS and LIST replies must only be used for CAP 302

jwheare opened the pull request:
Attempt to address #405

This version doesn't allow multiline LS replies to be sent to 301 clients. An alternative version that does might look like:

> Servers MAY send multiple lines in response to `CAP LS` and `CAP LIST` but SHOULD NOT send them to clients that do not support CAP version `302`. Clients that support CAP version `302` MUST and other clients SHOULD handle the continuation format described as follows:

But this seems odd. Why should 301 clients support this if servers shouldn't be sending it?

grawity commented:
I have to guess that clients "SHOULD" understand them because they don't know whether they're talking to a 301 server or to a server implementing [the draft spec][1] (which _did_ have multiline responses).

Of course the server also doesn't know whether it's replying to a 301 client or a draft-mitchell client, so I agree it should be conservative and only send a single line.

[1]: http://web.archive.org/web/20140728040332/http://www.leeh.co.uk/draft-mitchell-irc-capabilities-02.html

jwheare commented:
Yeah it's a case of "gee it would be nice if every client did this then we could go back to the mitchell spec for all". But I suspect the non-mitchell behaviour has been published for long enough now that that ship has sailed. So I think I'd rather just have a current spec that doesn't raise more questions than it answers.

lel-amri commented:
@jwheare I'd keep the examples for single-line CAP LS/LIST replies while modifying them so they show only one-lined replies.

jwheare commented:
Not sure that's necessary. Ordinary LS/LIST examples are already shown elsewhere

jwheare commented:
Dunno why "Version" was being capitalised all over the place, but I fixed that and rewrote the start of the multiline bit to match the language of the cap values bit.
